README.md

```diff
@@ -17,73 +38,81 @@ so the host part of the URL (probably) is not the real destination.
-The extension can keep this keys and configurations on the side of the user
+The extension can keep these keys and configurations on the side of the user
 and it does not expose the secrets to the applications that initialize the call.
 In this sense it also works like a `proxy`.
 
-During the redirect, the extension can create separated context for `root content IDs` and perform the action according to that.
+During the redirection, the extension creates separated context by `root content IDs` and performs the action according to that.
 (E.g. restrict calls towards the targeted service, cache the returned cookies from the response in the content context, etc.)
-This architecture also allows for used decentralized services to change its default URLs to any arbitrary one,
+This architecture also allows changing the default URLs of decentralized services (Bee) to any arbitrary one,
 meanwhile dApps do not have to guess this address.
 For example Bee client has default `http://localhost:1633`, user can change it to any other port or even other gateway host,
 the dApps will call it in the same way.
 
 ### Callable Endpoints
 
-For any action the Fake URL host is `http://localhost:1633`.
+For any action, the Fake URL host is `http://localhost:1633`.
 As it is written earlier, it is not the address of the Bee client,
 it is just a reserved host address that the extension took and built its Fake URL paths on it.
-If the user changes their Bee API address, these callings still remain the same from dApp side.
+If the user changes their Bee API address, these endpoints still remain the same from dApp side.
 
 - `http://localhost:1633/fake-url/bzz/*` - `BZZ protocol` redirects to this fake URL, thereby in `*` can be not only content reference with its path, but any BZZ protocol compatible reference. It will be redirected to `bzz` endpoint of the Bee client.
 - `http://localhost:1633/fake-url/bee-api/*` - it will be forwarded to the Bee API. `*` has to be valid Bee API path
 
 ## Custom Protocol
 
-Only supported Swarm protocol currently is `bzz`.
-It makes a redirect to the BZZ endpoint of the Bee node by calling its corresponding Fake URL.
-There will be need for other Swarm specific protocols (or extend the current one), which handles different type of feeds and mutable content.
+The Swarm protocol to address other P2P content is `bzz`. It makes a redirection to the BZZ endpoint of the Bee node.
+If you type `bzz://{content-address}` into the address bar, the page will be redirected to `http(s)://{your-bzz-node-host}/bzz/{content-address}`. Currently it only reacts like this if the default search engine of the browser is set to Google.
+It also behaves the same on simple google searches on `https://google.com`.
+
+Additionally, The extension currently [injects a script](src/contentscript/index.ts) on document load so that dApp pages could refer any other P2P resources.
```

Comment:
```suggestion
Additionally, the extension currently [injects a script](src/contentscript/index.ts) on document load so that dApp pages could refer any other P2P resources.
```